modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/db/Migration.scala

```diff
@@ -247,8 +247,8 @@ object Migration {
             val message = s"Table's schema key '${current.toSchemaUri}' cannot be found in fetched schemas $state. Migration cannot be created"
             DiscoveryFailure.IgluError(message).toLoaderError.asLeft
         }
-      case s: DSchemaList.Single =>
-        val message = s"Illegal State: updateTable called for a table with known single schema [${s.schema.self.schemaKey.toSchemaUri}]\ncolumns: ${columns.mkString(", ")}\nstate: $state"
+      case _: DSchemaList.Single =>
+        val message = s"Illegal State: updateTable called for a table with known single schema [${current.toSchemaUri}]\ncolumns: ${columns.mkString(", ")}\nstate: $state"
```

Comment:
@chuwy I just changed the `s.schema.self.schemaKey` to `current` as you described in the issue. Couldn't be sure what to do improve it further. If you have any idea, just let me know.

Reply:
Do you think we can change `Schema` case class there to JSON as well?

Reply:
I couldn't find where `Schema` case class is used in here. Could you show me, please ?

Reply:
We realized that `Schema` case class of SchemaDDL has derived encoder and it outputs lots of null values. We decided with @chuwy that best way is updating encoder in SchemaDDL initially then use it in here. Related issue on SchemaDDL can be found in here: https://github.com/snowplow/schema-ddl/issues/105